core/src/connection.rs

```diff
@@ -133,6 +133,13 @@ impl ConnectedPoint {
         }
     }
 
+    pub fn get_remote_address(&self) -> &Multiaddr {
```

Comment:
A short doc-comment may be good, possibly with a hint / reminder that this address may be "private" to the connection, i.e. may not be a listening address of the remote and hence not usable for establishing new connections.